google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/application/DefaultAppEngineAdminService.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.intellij.appengine.application;
+
+import com.google.api.client.auth.oauth2.Credential;
+import com.google.api.client.googleapis.json.GoogleJsonResponseException;
+import com.google.api.services.appengine.v1.model.Application;
+import com.google.api.services.appengine.v1.model.ListLocationsResponse;
+import com.google.api.services.appengine.v1.model.Location;
+import com.google.api.services.appengine.v1.model.Operation;
+import com.google.cloud.tools.intellij.appengine.cloud.AppEngineOperationFailedException;
+import com.google.cloud.tools.intellij.resources.GoogleApiClientFactory;
+
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.io.IOException;
+import java.util.List;
+
+public class DefaultAppEngineAdminService extends AppEngineAdminService {
+
+  private final static String APP_ENGINE_RESOURCE_WILDCARD = "-";
+  private final long CREATE_APPLICATION_POLLING_INTERVAL_MS = 1000;
+
+  @Override
+  @Nullable
+  public Application getApplicationForProjectId(@NotNull String projectId,
+      @NotNull Credential credential) throws IOException {
+    try {
+      return GoogleApiClientFactory.getInstance().getAppEngineApiClient(credential)
+          .apps().get(projectId).execute();
+    } catch (GoogleJsonResponseException e) {
+      if (e.getStatusCode() == 404) {
+        // the application does not exist
+        return null;
+      }
+      throw e;
```

Comment:
Not sure I understand what you're asking. Are you suggesting wrapping this in an unchecked exception or something like that? Or are you suggesting declaring `GoogleJsonReponseException` in the throws clause of the method signature?

FYI GoogleJsonResponseException inherits from IOException, so this is a way of allowing callers to deal with a transient exception (i.e. perhaps to retry).